src/Microsoft.Azure.WebJobs.Host/JobHost.cs

```diff
@@ -23,7 +24,7 @@ namespace Microsoft.Azure.WebJobs
     /// A <see cref="JobHost"/> is the execution container for jobs. Once started, the
     /// <see cref="JobHost"/> will manage and run job functions when they are triggered.
     /// </summary>
-    public class JobHost : IDisposable
+    public class JobHost : IDisposable, IServiceProvider
```

Comment:
Why are we making the `JobHost` an `IServiceProvider`? Based on our chat, this shouldn't be required to address the existing requirements for on-prem, and conflicts with the DI design plans we've discussed. 
`JobHost` is already doing more than it should as-is, it would be nice to avoid taking on this additional responsibility here. 

Can we leverage the existing model/support until we have the proper DI implementation in place?